Marlin/src/lcd/ultralcd.cpp

```diff
@@ -3742,22 +3742,37 @@ void kill_screen(const char* lcd_msg) {
      * "Print from SD" submenu
      *
      */
+
     #if ENABLED(SD_REPRINT_LAST_SELECTED_FILE)
       uint32_t saved_encoderPosition = 0;
-      static millis_t assume_print_finished = 0;
+
+      void lcd_reselect_last_file() {
+        // Some of this is a hack to force the screen update to work.
+        // TODO: Fix the real issue that causes this!
+        lcdDrawUpdate = LCDVIEW_CALL_REDRAW_NEXT;
+        _lcd_synchronize();
+        safe_delay(50);
+        _lcd_synchronize();
+        lcdDrawUpdate = LCDVIEW_CALL_REDRAW_NEXT;
+        drawing_screen = screen_changed = true;
+        lcd_goto_screen(lcd_sdcard_menu, saved_encoderPosition);
+        defer_return_to_status = true;
+        lcd_update();
+      }
     #endif
 
     void lcd_sdcard_menu() {
       ENCODER_DIRECTION_MENUS();
 
       #if ENABLED(SD_REPRINT_LAST_SELECTED_FILE)
+        static millis_t assume_print_finished = 0;
         if (ELAPSED(millis(), assume_print_finished)) { // if the printer has been busy printing, lcd_sdcard_menu() should not
           lcdDrawUpdate = LCDVIEW_REDRAW_NOW;           // have been active for 5 seconds.  In this case, restore the previous
           encoderPosition = saved_encoderPosition;      // encoderPosition to the last selected item.
           assume_print_finished = millis() + 5000;
         }
```

Comment:
Note that this will fire always right away if you leave the SD menu for 5s or more and then return to it again, and will fire every 5s while inside the SD menus.

If it only needs to fire the first time when returning to the SD menu after a print, instead of a timer, a simple flag set on Start SD Print should be sufficient.

Reply:
I'll have to test and see what happens if you remove the whole block from the function. Seems like everything should still work… Also, it seems like it would be best in `lcd_reselect_last_file` to call `lcd_goto_screen` at the very top, and only afterward force at least one screen update.